src/modules/cpu/kernel/normalize.hpp

```diff
@@ -0,0 +1,340 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+void compute_diff_square_sum(Rpp32f &output, Rpp32f *input, Rpp64s inputStride, Rpp64s numElements, Rpp32f mean)
+{
+    const Rpp64s stride = 1;
+    if (numElements > 32)
+    {
+        Rpp64s currElements = numElements >> 1;
+        Rpp32f tmp1 = 0, tmp2 = 0;
+
+        // reduce first half and accumulate
+        compute_diff_square_sum(tmp1, input, stride, currElements, mean);
+
+        // reduce second half and accumulate
+        compute_diff_square_sum(tmp2, input + currElements * stride, stride, numElements - currElements, mean);
+
+        tmp1 += tmp2;
+        output += tmp1;
+    }
+    else
+    {
+        // reduce to a temporary
+        Rpp32f tmp = 0;
+        for (Rpp64s i = 0; i < numElements; i++)
+        {
+            Rpp32f curr = (input[i * stride] - mean);
+            auto curnew = curr * curr;
+            tmp += curnew;
+        }
+
+        // accumulate in target value
+        output += tmp;
+    }
+}
+
+void compute_sum(Rpp32f &output, Rpp32f *input, Rpp64s inputStride, Rpp64s numElements)
+{
+    const Rpp64s stride = 1;
+    if (numElements > 32)
+    {
+        Rpp64s currElements = numElements >> 1;
+        Rpp32f tmp1 = 0, tmp2 = 0;
+
+        // reduce first half and accumulate
+        compute_sum(tmp1, input, stride, currElements);
+
+        // reduce second half and accumulate
+        compute_sum(tmp2, input + currElements * stride, stride, numElements - currElements);
+
+        tmp1 += tmp2;
+        output += tmp1;
+    }
+    else
+    {
+        // reduce to a temporary
+        Rpp32f tmp = 0;
+        for (Rpp64s i = 0; i < numElements; i++)
+            tmp += input[i * stride];
+
+        // accumulate in target value
+        output += tmp;
+    }
+}
+
+Rpp32f rpp_rsqrt(Rpp32f x)
+{
+    // Use SSE intrinsic and one Newton-Raphson refinement step
+    // - faster and less hacky than the hack below.
+    __m128 X = _mm_set_ss(x);
+    __m128 tmp = _mm_rsqrt_ss(X);
+    Rpp32f y = _mm_cvtss_f32(tmp);
+    return y * (1.5f - x * 0.5f * y * y);
+}
+
+static void rpp_rsqrt_sse(Rpp32f *input, Rpp64s numElements, Rpp32f eps, Rpp32f rdiv, Rpp32f mul)
+{
+    Rpp64s i = 0;
+    __m128 rdivx4 = _mm_set1_ps(rdiv);
+    __m128 mulx4 = _mm_set1_ps(mul * 0.5f);
+    if (eps) // epsilon is present - no need for masking, but we need to add it
+    {
+        __m128 epsx4 = _mm_set1_ps(eps);
+        for (; i + 4 <= numElements; i += 4)
+        {
+            __m128 x = _mm_loadu_ps(&input[i]);
```

Comment:
Please try to change all this code from SSE to AVX2 and check if this gives any improvement

Reply:
Yeah with this change seeing 2% improvement, will incorporate this change.